AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStoreAdmin/src/InitializeMutation.dfy

```diff
@@ -195,14 +196,24 @@ module {:options "/functionSyntax:4" } InternalInitializeMutation {
       Types.KeyStoreAdminException(
         message := "Active Branch Key Item read from storage is malformed!")
     );
-
-      // TODO-HV-2-M2: Support items in HV-2
+    // TODO-HV-2-M2: Support items in HV-2
     :- Need(
       readItems.ActiveItem.EncryptionContext[Structure.HIERARCHY_VERSION] == Structure.HIERARCHY_VERSION_VALUE_1,
       Types.KeyStoreAdminException(
         message := "At this time, Mutations ONLY support HV-1; BK's Active Item is HV-2.")
     );
-
+    if (
+        && input.Mutations.TerminalHierarchyVersion.Some?
+        && input.Mutations.TerminalHierarchyVersion.value.v2?
+      ) {
+      :- Need(
+        HvUtils.HasUniqueTransformedKeys?(readItems.ActiveItem.EncryptionContext),
+        Types.UnexpectedStateException(
+          message :=
+            KeyStoreErrorMessages.NOT_UNIQUE_BRANCH_KEY_CONTEXT_KEYS
+        )
```

Comment:
Blocking: I think I want to call this a user error, 
as they MUST have modified the BK without using the library.
In which case, this would be KeyStoreAdminException
```suggestion
      :- Need(
        HvUtils.HasUniqueTransformedKeys?(readItems.ActiveItem.EncryptionContext),
        Types.KeyStoreAdminException(
          message :=
            KeyStoreErrorMessages.NOT_UNIQUE_BRANCH_KEY_CONTEXT_KEYS
        )
```